### PR TITLE
[mono] Improve RuntimeConstructorInfo.ToString

### LINKED ISSUE #36688: Mono: ConstructorInfo.ToString () returns unexpected format
Example:

```csharp
public class Application
{
	static void Main (string[] args)
	{
		Console.WriteLine (typeof (Application).GetConstructors ()[0]);
	}
}
```

Actual result:

```
.ctorVoid ()
```

Expected result (output when using mono or a dotnet console app):

```
Void .ctor()
```

### PR DESCRIPTION
**Pull request by alexischr**
Use CoreCLR's code. Fixes https://github.com/dotnet/runtime/issues/36688

### CODE REVIEW

File: src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs [hunk 1]
```diff
@@ -751,6 +751,7 @@ internal class RuntimeConstructorInfo : ConstructorInfo
 #pragma warning disable 649
         internal IntPtr mhandle;
         private string? name;
+        private string? toString;
         private Type? reftype;
```
Comment (vargaz) [inline src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:755]: The runtime depends on this layout, so if new fields are added, they should be added at the end.

Comment (alexischr) [reply src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:755]: That explains all the crashes :) I will add a comment to the class

Comment (vargaz) [reply src/mono/netcore/System.Private.CoreLib/src/System/Reflection/RuntimeMethodInfo.cs:754]: We usually add something like:
```
		#region Sync with ReflectionAssembly in object-internals.h
...
                 #endregion
```

**Comment by alexischr** [comment]
@vargaz Added the `sync` comment

**Comment by SamMonoRT** [comment]
@vargaz - please review again when you have a moment.

**Review by akoeplinger — APPROVED** [review]
LGTM but would be nice to add a test, maybe somewhere in https://github.com/dotnet/runtime/blob/master/src/libraries/System.Reflection.TypeExtensions/tests/ConstructorInfo/ConstructorInfoTests.cs ?

**Comment by SamMonoRT** [comment]
Will create a work item to add a new test. We should merge this as is. @imhameed fyi.

**Pull request merged**